openmdao/core/problem.py

```diff
@@ -940,6 +940,14 @@ def setup(self, check=False, logger=None, mode='auto', force_alloc_complex=False
         # set static mode back to True in all systems in this Problem
         self._metadata['static_mode'] = True
 
+        des_vars = self.model.get_design_vars()
```

Comment:
Rather than doing this here and having a whole extra traversal down the system tree, it might be better to do this down in the `System._get_design_vars` method.

Reply:
I needed to do it in the setup function because the system level `_get_design_vars` method loops through each subsystem and I need the final values all at once.